src/iqm/iqm_client/transpile.py

```diff
@@ -0,0 +1,392 @@
+    def apply_move(self, qubit: str, resonator: str):
+        """Apply the logical changes of the resonator state location when a move gate between qubit and resonator is
+        applied.
+
+        Args:
+            qubit (str): The moved qubit.
+            resonator (str): The moved resonator.
+
+        Raises:
+            CircuitExecutionError: When the move is not allowed, either because the resonator does not exist, the move
+            gate is not valid between this qubit-resonator pair, or the resonator state is currently in a different
+            qubit register.
+        """
+        if (
+            resonator in self.resonators
+            and qubit in self.move_calibrations[resonator]
+            and self.res_qb_map[resonator] in [qubit, resonator]
+        ):
+            self.res_qb_map[resonator] = qubit if self.res_qb_map[resonator] == resonator else resonator
+        else:
+            raise CircuitExecutionError('Attempted move is not allowed.')
+
+    def create_move_instructions(
+        self,
+        qubit: str,
+        resonator: str,
+        apply_move: Optional[bool] = True,
+        alt_qubit_names: Optional[dict[str, str]] = None,
+    ):
+        """Create the move instructions needed to move qubit and resonator.
+
+        Args:
+            qubit (str): The qubit
+            resonator (str): The resonator
+            apply_move (Optional[bool], optional): Whether the moves should be applied to the resonator tracking state.
+            Defaults to True.
+
+        Yields:
+            Instruction: The one or two move instructions needed.
+        """
+        if self.res_qb_map[resonator] not in [qubit, resonator]:
+            other = self.res_qb_map[resonator]
+            if apply_move:
+                self.apply_move(other, resonator)
+            qbs = tuple(alt_qubit_names[q] if alt_qubit_names else q for q in [other, resonator])
+            yield Instruction(name=self.move_gate, qubits=qbs, args={})
+        if apply_move:
+            self.apply_move(qubit, resonator)
+        qbs = tuple(alt_qubit_names[q] if alt_qubit_names else q for q in [qubit, resonator])
+        yield Instruction(name=self.move_gate, qubits=qbs, args={})
+
+    def reset_as_move_instructions(
+        self,
+        resonators: Optional[Iterable[str]] = None,
+        apply_move: Optional[bool] = True,
+        alt_qubit_names: Optional[dict[str, str]] = None,
+    ):
+        """Creates the move instructions needed to move all resonator states to their original state.
+
+        Args:
+            resonators (Optional[Iterable[str]], optional): The set of resonators to reset, if None, all resonators will
+            be reset. Defaults to None.
+            apply_move (Optional[bool], optional): Whether the moves should be applied to the resonator tracking state.
+            Defaults to True.
+
+        Returns:
+            list[Instruction]: The instructions needed to move all qubit states out of the resonators.
```

Comment:
```suggestion
            resonators: The set of resonators to reset, if None, all resonators will
            be reset.
            apply_move: Whether the moves should be applied to the resonator tracking state.

        Returns:
            The instructions needed to move all qubit states out of the resonators.
```
The types and defaults can be seen from the signature which is automatically included in the API docs.